volumes.cpp

```diff
@@ -163,6 +163,258 @@ class PlaneSurface : public SurfaceVolume {
 
 typedef  enum { X=0, Y=1, Z=2 } axis_t;
 
+class GeneralQuadraticSurface : public SurfaceVolume {
+
+protected:
+  double A_,B_,C_,D_,E_,F_,G_,H_,J_,K_;
+  int type;
+  Vector3d translation;
+  double rotation_mat[9];
+  double extents[3];
+  double gq_tol = 1e-6;
```

Comment:
It certainly can be for now, but from @makeclean's comments it looks like it might be an externally specified parameter in the future.